content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -0,0 +1,183 @@
+---
+title: 'Create custom Prisma Client queries'
+metaTitle: 'Prisma Client extensions: query component (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+tocDepth: 3
+---
+
+<TopBlock>
+
+You can use the `query` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component type to hook into the query life-cycle and modify an incoming query or its result. We introduced this feature in version 4.7.0.
+
+You can use the `query` extension component as an alternative to [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases). Unlike middlewares, the `query` extension component gives you end-to-end type safety. You can also use it to provide [user isolation](/concepts/components/prisma-client/client-extensions#extended-clients) in a row-level security (RLS) scenario. [Learn more](#query-extensions-versus-middlewares).
```

Comment:
(I feel like i left such a comment before...) The RLS example here is a bit strange, as "user isolation" is a specific property of that implementation and now a general thing that the query extension allows.